src/OpenTelemetry.Exporter.Zipkin/ZipkinExporterHelperExtensions.cs

```diff
@@ -38,22 +38,35 @@ public static TracerProviderBuilder AddZipkinExporter(this TracerProviderBuilder
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            var exporterOptions = new ZipkinExporterOptions();
-            configure?.Invoke(exporterOptions);
-            var zipkinExporter = new ZipkinExporter(exporterOptions);
+            if (builder is IDeferredTracerBuilder deferredTracerBuilder)
+            {
+                return deferredTracerBuilder.Configure((sp, builder) =>
+                {
+                    AddZipkinExporter(builder, sp.GetOptions<ZipkinExporterOptions>(), configure);
+                });
+            }
+
+            return AddZipkinExporter(builder, new ZipkinExporterOptions(), configure);
```

Comment:
I personally don't like how this approach is causing all exporters and instrumentations to have to type check the builder and then have different code paths. IMO, it shouldn't be optional for these exporters and instrumentations to support separating configuration and construction phases. It should be mandatory that they register things with a deferred approach so that we can keep configuration and construction phases cleanly separated.

Reply:
I would personally prefer if we could force all of them to be registered with lambdas and not even make it an option to immediately new them up when configuring.

Reply:
Of everything on this PR, I'm the least happy with this bit. But I don't think it is possible to do it any other way 😭 It's not really a problem of separating configuration and construction, this is about the inversion of dependencies we've created for ourselves. To do it "properly" we would need this ctor: `public ZipkinExporter(IOptions<ZipkinExporterOptions> options)`. But that of course requires a direct dependency.